R/method-correlation.R

```diff
@@ -59,9 +59,205 @@ setMethod("correlation", signature("data.table", "data.table"), function(data1,
 
 })
 
+#' Correlation
+#'
+#' This function returns correlation coefficients for all columns in one data table against themselves.
+#' 
+#' @param data1 data.table with columns as variables. All columns must be numeric. One row per sample.
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return data.frame with correlation coefficients
+setMethod("correlation", signature("data.table", "missing"), function(data1, data2, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) {
```

Comment:
Based on the tests it looks like this method returns all n x n correlation pairs (n = number columns in data1). But, the resulting correlation matrix is symmetric, so we don't need all of these results. In fact, if we assume all correlations of columns against themselves are 1 (as they must be), then we should return only n(n-1)/2 correlation coefficients. I don't see any arguments to `cor` that get at this, but we could just remove duplicate rows of `corrResult` and any row with `data1` = `data2` (a self correlation).

Reply:
im not sure if removing dup rows helps, bc presumably the source, target/ data1, data2 cols will be reversed for the dup.. if this were a matrix i think we could use `lower.tri`. ill think about it some.